liberty-maven-plugin/src/main/java/net/wasdev/wlp/maven/plugins/server/DevMojo.java

```diff
@@ -418,7 +318,28 @@ public void recompileBuildFile(File buildFile, List<String> artifactPaths) {
                 log.debug("Could not recompile pom.xml", e);
             }
         }
+        
+        public int getMessageOccurrences(String regexp, File logFile) {
+            int messageOccurrences = -1;
+            try {
+                logFile = serverTask.getLogFile();
+                regexp = UPDATED_APP_MESSAGE_REGEXP + DevMojo.this.project.getArtifactId();
+                messageOccurrences = serverTask.countStringOccurrencesInFile(regexp, logFile);
+                log.debug("Message occurrences before compile: " + messageOccurrences);
+            } catch (Exception e) {
+                log.debug("Failed to get message occurrences before compile", e);
+            }
+            return messageOccurrences;
 
+        }
+        
+        public void runTestThread(ThreadPoolExecutor executor, String regexp, File logFile, int messageOccurrences) {
```

Comment:
As a best practice, add the `@Override` annotation since it is implementing an abstract method.